### PR TITLE
Pre-release zkSync Wallet v.2.4.0

### PR DESCRIPTION
**Pull request by keriat**

### CODE REVIEW

File: src/store/account.ts [hunk 1]
```diff
@@ -45,6 +45,7 @@ export const mutations = mutationTree(state, {
     if (state.address !== undefined) {
       if (name.length < 1) {
         name = getNameFromAddress(state.address);
+        window.localStorage.removeItem(state.address);
```
Comment (StanislavBreadless) [inline src/store/account.ts:48]: Why do we save the address in the first place?

File: src/store/wallet.ts [hunk 2]
```diff
@@ -231,28 +212,31 @@ export const actions = actionTree(
...
      */
-    async requestZkBalances({ commit, getters }, { accountState, force = false }) {
+    async requestZkBalances({ state, commit, getters }, { accountState, force = false }) {
       type BalancesList = {
         [token: string]: BigNumberish;
       };
-      let listCommitted = <BalancesList>{};
-      let listVerified = <BalancesList>{};
-      const tokensList = <Array<ZkInBalance>>[];
+      let listCommitted: BalancesList = <BalancesList>{};
```
Comment (StanislavBreadless) [inline src/store/wallet.ts:235]: You need to specify the type only once

File: tests/store/account.spec.ts [hunk 3]
```diff
@@ -81,3 +81,6 @@ describe("Account getters", () => {
     });
   });
 });
+function beforeEach(arg0: () => void) {
```
Comment (StanislavBreadless) [inline tests/store/account.spec.ts:84]: Is this function ever used?

Mocha `beforeEach` hook is usually specified  inside the `describe`

**Review by StanislavBreadless — COMMENTED** [review]
Most of my comment are code-style nits, but some of them are related to `cpk` pre-signing logic.

**Comment by StanislavBreadless** [comment]
@keriat 

Does this PR include speed optimizations?

File: .github/workflows/firebase-hosting-push-master.yml [hunk 4]
```diff
@@ -21,29 +21,15 @@ jobs:
...
       - name: Deploy to rinkeby site
         uses: matter-labs/action-hosting-deploy@v0
         with:
-          firebaseServiceAccount: '${{ secrets.FIREBASE_SERVICE_ACCOUNT_ZKSYNC_VUE }}'
+          firebaseServiceAccount: "${{ secrets.FIREBASE_SERVICE_ACCOUNT_ZKSYNC_VUE }}"
           target: rinkeby
           projectId: zksync-vue
           channelID: live
 
-      - name: Configure for ropsten
```
[the remark this answers is not in the document]
Comment (keriat) [reply .github/workflows/firebase-hosting-push-master.yml:33]: It's probably redundant, cause I've re-configured CI.

File: .github/workflows/firebase-hosting-push-dev.yml [hunk 5]
```diff
@@ -2,58 +2,58 @@
...
-      - name: Configure for mainnet [dev]
+      - name: "[mainnet-dev] Deploy to mainnet site"
         run: bash ./cli-process-env.sh mainnet dev 1
 
-      - name: Build mainnet
+      - name: "[mainnet-dev] Building mainnet"
         run: yarn generate --fail-on-error
 
-      - name: Deploy to mainnet [dev]
+      - name: Deploy to ropsten site
```
[the remark this answers is not in the document]
Comment (keriat) [reply .github/workflows/firebase-hosting-push-dev.yml]: Redundant, CI was reconfigured

File: .github/workflows/firebase-beta-ci.yml [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+        run: bash ./cli-process-env.sh ropsten prod 1
+
+      - name: "[Ropsten release stage] Step 2: Building ropsten"
+        run: yarn generate --fail-on-error
+
+      - name: "[Ropsten release stage] Step 3: Publishing ropsten"
+        uses: matter-labs/action-hosting-deploy@v0
+        with:
+          firebaseServiceAccount: "${{ secrets.FIREBASE_SERVICE_ACCOUNT_ZKSYNC_VUE }}"
+          target: ropsten
```
[the remark this answers is not in the document]
Comment (keriat) [reply .github/workflows/firebase-beta-ci.yml]: I've discussed with @Deniallugo and if i got things right for some time they will be both linked with the beta

File: src/pages/deposit.vue [hunk 7]
```diff
@@ -363,32 +372,33 @@ export default Vue.extend({
...
-        this.transactionInfo.fee = {
-          amount: receipt.gasUsed.toString(),
-          token: ETHToken,
-        };
+        this.transactionInfo.fee = undefined;
+        this.tokenAllowance = await this.getTokenAllowance(this.chosenToken);
         this.transactionInfo.continueBtnFunction = true;
-        this.transactionInfo.continueBtnText = "Proceed to deposit";
+        if (!unlimited) {
+          this.transactionInfo.continueBtnText = "Proceed to deposit";
```
[the remark this answers is not in the document]
Comment (JackHamer09) [reply src/pages/deposit.vue:394]: If you unlock unlimited amount then we don't care if user enters the different amount and he doesn't even need to enter any amount to unlock unlimited (so he can't proceed to deposit after unlocking).
But if he enters amount then he can unlock this specific amount and we don't really want him to change deposit amount to not confuse him, overwise he will have to unlock again if he enters bigger amount (but there is still "go back" button tho)

File: nuxt.config.ts [hunk 8]
```diff
@@ -238,22 +228,36 @@ const config: NuxtConfig = {
...
+    extend: (config: Configuration) => {
       config.node = {
         fs: "empty",
       };
     },
   },
-  generate: {
-    dir: "public",
+  buildOptimisations: {
+    profile: env !== "prod" ? "risky" : "experimental",
```
[the remark this answers is not in the document]
Comment (JackHamer09) [reply nuxt.config.ts:247]: I think it's actually "risky" when NOT production

File: src/components/Transaction.vue [hunk 9]
```diff
@@ -94,7 +96,7 @@
                 {{ feeToken.symbol }}
               </span>
               <span class="totalPrice">
-                {{ feesObj.normal | formatUsdAmount(feeToken.tokenPrice, feeToken.symbol) }}
+                <token-price :symbol="feeToken.symbol" :amount="feesObj.normal.toString()" />
```
[the remark this answers is not in the document]
Comment (JackHamer09) [reply src/components/Transaction.vue:99]: It's radio input for fast and normal withdraw which is temporarily disabled.

File: src/plugins/walletActions/transaction.ts [hunk 10]
```diff
@@ -22,43 +24,45 @@ export const transaction = async (
...
   const syncWallet = walletData.get().syncWallet;
+  const nonce = await syncWallet!.getNonce("committed");
+  const batchBuilder = syncWallet!.batchBuilder(nonce);
 
-  /**
-   * @todo: process case when there are 2 transactions
-   */
+  if (store.wallet.isAccountLocked) {
+    if (!accountActivationFee) {
+      throw new Error("No account activation fee found");
```
[the remark this answers is not in the document]
Comment (JackHamer09) [reply src/plugins/walletActions/transaction.ts:35]: Didn't get what you mean

File: src/store/account.ts (hunk 1 above)
Comment (JackHamer09) [reply src/store/account.ts:48]: It's an account name, which the user can change.

File: tests/store/account.spec.ts (hunk 3 above)
Comment (JackHamer09) [reply tests/store/account.spec.ts:84]: not even sure we use this tests, I think they are from the older versions and actually almost useless

File: src/blocks/modals/SignPubkey.vue [hunk 11]
```diff
@@ -0,0 +1,124 @@
...
+          if (!isOnchainAuthSigningKeySet) {
+            const onchainAuthTransaction = await syncWallet!.onchainAuthSigningKey();
+            await onchainAuthTransaction?.wait();
+          }
+        }
+
+        const ethAuthType = syncWallet?.ethSignerType?.verificationMethod === "ERC-1271" ? "Onchain" : "ECDSA";
+        let changePubKeyMessage;
+        const newPubKeyHash = await syncWallet!.signer!.pubKeyHash();
+        if (ethAuthType === "ECDSA") {
```
[the remark this answers is not in the document]
Comment (StanislavBreadless) [reply src/blocks/modals/SignPubkey.vue:78]: No

File: src/plugins/walletActions/transaction.ts (hunk 10 above)
[the remark this answers is not in the document]
Comment (StanislavBreadless) [reply src/plugins/walletActions/transaction.ts:35]: Dunno, probably when I looked at this there was a different error message :thinking:

File: tests/store/account.spec.ts (hunk 3 above)
Comment (StanislavBreadless) [reply tests/store/account.spec.ts:84]: Maybe remove it then?

File: README.md [hunk 12]
```diff
@@ -61,15 +54,44 @@ For detailed explanation on how things work, check out [Nuxt.js docs](https://nu
...
+For detailed explanation on how things work, check out [Nuxt.js docs](https://nuxtjs.org).
+
+---
+
+# zkSync Ecosystem
+
+- [Integration Docs](https://zksync.io/dev)
+- [Available API & protocols](https://zksync.io/api/)
+- [zkWallet](https://wallet.zksync.io/)
+- [zkMint](https://mint.zksync.dev/)
```
[the remark this answers is not in the document]
Comment (StanislavBreadless) [reply README.md:97]: `https://mint.zksync.dev`

When I was talking to Alex he told me that all the public websites should be `.io`. We just need to discuss this together with him

**Review by StanislavBreadless — APPROVED** [review]

**Pull request merged**